### ISSUE TITLE
No citation found

### ISSUE THREAD

**Issue by karimmd** [mgmeyers/obsidian-pandoc-reference-list#83]
The installed **Pandoc Reference List** is not working for me. I am not sure whether I have made any mistakes while configuring it. 

This is the issue that I am having while trying to import markdown citation using **Citations** plugins.
![image](https://github.com/mgmeyers/obsidian-pandoc-reference-list/assets/6984392/6e1da500-879a-4bea-bbfc-9fec77b12a47)

Here's my configuration for **Pandoc Reference List** within Obsidian
![image](https://github.com/mgmeyers/obsidian-pandoc-reference-list/assets/6984392/a9797f66-916d-4b72-9082-0aeb6f45f2eb)

Here's the configuration from **Citations** plugins within Obsidian
![image](https://github.com/mgmeyers/obsidian-pandoc-reference-list/assets/6984392/a8cb2f03-f15e-4dd7-a937-90054a60cb62)

Here's the configuration from Zotero
![image](https://github.com/mgmeyers/obsidian-pandoc-reference-list/assets/6984392/92eba74a-44eb-4edf-9b40-abd0a05b5d7c)

My question is what I am missing here?

**Comment by napseis** [comment]
hi, do you have this error too ?

node:events:491 Uncaught Error: getaddrinfo ENOTFOUND raw.githubusercontent.com
    at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
    at __node_internal_ (node:internal/errors:712:10)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:108:26)
__node_internal_captureLargerStackTrace	@	node:internal/errors:490
__node_internal_	@	node:internal/errors:712
onlookup

**Comment by karimmd** [comment]
I do not have this error.

**Comment by napseis** [comment]
It should pop up in the console (ctrl + shift + i )

**Comment by mgmeyers** [comment]
Yes, please check the console for errors. The plugin might be having problems reading your bibliography file

**Comment by napseis** [comment]
it seems to me it's a proxy problem. For some reason raw.githubusercontent.com is not accessible on the command line.
I'm in a company, I had to configure git to use a proxy, the system (windows...) one was ignored. Is there a workaround ?

**Comment by HansonChongZZ** [comment]
I am having the same issue on MacOS. Any updates on solutions? 

![image](https://github.com/mgmeyers/obsidian-pandoc-reference-list/assets/114476269/28cee60e-0707-4b26-ac5b-0a05cf807967)

Here are my settings: 
<img width="601" alt="image" src="https://github.com/mgmeyers/obsidian-pandoc-reference-list/assets/114476269/b531e811-fe59-41fc-a827-20b70056602b">

**Comment by pullopen** [comment]
I also have the same problem.

**Comment by pullopen** [comment]
> hi, do you have this error too ?
> 
> node:events:491 Uncaught Error: getaddrinfo ENOTFOUND raw.githubusercontent.com at __node_internal_captureLargerStackTrace (node:internal/errors:490:5) at __node_internal_ (node:internal/errors:712:10) at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:108:26) __node_internal_captureLargerStackTrace @ node:internal/errors:490 __node_internal_ @ node:internal/errors:712 onlookup

Same here. Any solution?

**Comment by egouldo** [comment]
Does the error resolve if using a custom citation style language?

**Comment by napseis** [comment]
Actually I tried it last friday: downloaded the file via the web browser, gave it as a personalized style, but still try to connect. Tried to mess up the plugin too, to no avail, I don't know java and the file is not structured.

**Comment by jmiba** [comment]
I have the error if I specify the language in the front matter. Console:
plugin:obsidian-pandoc-reference-list:79 TypeError: o.trim is not a function
    at QL (plugin:obsidian-pandoc-reference-list:79:4649)
    at hd.eval (plugin:obsidian-pandoc-reference-list:79:11643)
    at Generator.next (<anonymous>)
    at n (plugin:obsidian-pandoc-reference-list:6:1288)

**Comment by miyuchina** [comment]
I fixed my issue by:

* Using a custom citation style, and
* [Manually downloading my CSL locale file](https://github.com/citation-style-language/locales) and placing it in `vault/.pandoc/locales-en-US.xml`.

It does seem that the problem is network-related. I tracked down the error to `getCSLLocale` and `getCSLStyle`. It's the `https.get` that's emitting the error:

https://github.com/mgmeyers/obsidian-pandoc-reference-list/blob/d9ef5e5881b02b21c9aa75f30306fd3fdb2542df/src/bib/helpers.ts#L92-L110

I wonder if it's because `https.get` is not picking up my system proxy information. If so, I have no idea how to fix it. Replacing these calls with the `requestUrl` method [provided by Obsidian's API](https://docs.obsidian.md/Reference/TypeScript+API/requestUrl) doesn't appear to work any better — but I'm unfamiliar with the API in general, and maybe I'm doing something wrong.

@mgmeyers: As a temporary measure, is it possible to add a "Custom citation style language" to the settings tab (similar to "Custom citation style") and allow users to point to their own locales file? That way we can bypass the network requests altogether in case they don't work.

**Comment by ArendtSun** [comment]
Hi, today I set up the pandoc reference list and zotero7 according to the tutorial, and the csl format is united as APA7, and it also shows citation not found, which is similar to the problem you found, and I would like to know if this problem is solved later? How was it solved, was it because the APA format was not recognised?
